tools/setup_helpers/cmake.py

```diff
@@ -232,61 +232,73 @@ def generate(self, version, cmake_python_library, build_python, build_test, my_e
         _mkdir_p(install_dir)
         _mkdir_p(self.build_dir)
 
+        # Store build options that are directly stored in environment variables
+        build_options = {
+            # The default value cannot be easily obtained in CMakeLists.txt. We set it here.
+            'CMAKE_PREFIX_PATH': distutils.sysconfig.get_python_lib()
+        }
+        # Options that do not start with 'USE_' or 'BUILD_' and are directly controlled by env vars
+        additional_options = {
+            'BLAS',
+            'BUILDING_WITH_TORCH_LIBS',
+            'CMAKE_PREFIX_PATH',
+            'ONNX_ML',
+            'ONNX_NAMESPACE',
+            'WERROR'
+        }
+        for var, val in my_env.items():
+            # We currently pass over all environment variables that start with "BUILD_" or "USE_". This is because we
+            # currently have no reliable way to get the list of all build options we have specified in CMakeLists.txt.
+            # (`cmake -L` won't print dependent options when the dependency condition is not met.) We will possibly
```

Comment:
Alternately, does that mean we should just get rid of dependent options?

Reply:
Some more thinking: it's very useful to have some way of telling the user what all available options / environment variables are. Right now we just hand document a few notable ones in `setup.py` but if there were the moral equivalent of `--help` that seems useful for users. (Nothing to do for this PR)

Reply:
I am thinking about having all option descriptions in `CMakeLists.txt` in standard CMake way, and creating a method in a Python script extracting these descriptions. So long as we restrict ourselves to write these options (and their descriptions and dependency relationship) in a way that the to-be-created Python method can recognize (e.g., no super-convoluted CMake syntax playing), and I do not see a problem.